packages/components/icons/src/icon.story.js

```diff
@@ -25,6 +37,147 @@ const IconContainer = styled.div`
+          </RadioInput.Option>
+          {activeExample === 'dirtySvg' && (
+            <SelectInput
+              value={activeSvgFixture}
+              onChange={(event) => {
+                setActiveSvgFixture(event.target.value);
+              }}
+              options={svgFixtures.dirtySvg.map((fixture, index) => ({
+                label: fixture.title,
+                value: fixture.payload,
+              }))}
+              components={{
+                Option: SelectSvgFixtureOption,
+              }}
+            />
+          )}
+          <RadioInput.Option value="custom">
+            <Spacings.Inline alignItems="center">Custom SVG</Spacings.Inline>
+          </RadioInput.Option>
+          {activeExample === 'custom' && (
+            <MultilineTextInput
+              value={customSvgData}
+              onChange={(event) => {
+                setCustomSvg(event.target.value);
+              }}
+            />
+          )}
+        </RadioInput.Group>
+      </Spacings.Stack>
+      <Spacings.Stack scale="m">
+        <Text.Headline as="h2">Rendered:</Text.Headline>
+        {colorValues.map((color) => (
+          <Spacings.Stack scale="xs" key={color}>
+            <Text.Body isBold>
+              <code>color: {color}</code>
+            </Text.Body>
+            <div
+              style={{
+                height: '100%',
+                backgroundColor:
+                  color === 'surface' ? customProperties.colorSolid : 'inherit',
+              }}
+            >
+              <Grid
+                gridGap={customProperties.spacingS}
+                gridAutoColumns="1fr"
+                gridTemplateColumns={`repeat(${colorValues.length}, 1fr)`}
+                alignItems="center"
+              >
+                {sizeValues.map((size) => {
+                  let data = '';
+                  switch (activeExample) {
+                    case 'cleanSvg': {
+                      data = svgFixtures.cleanSvg;
+                      break;
+                    }
+                    case 'dirtySvg': {
+                      const fixture = svgFixtures.dirtySvg.find(
+                        (fixture) => fixture.payload === activeSvgFixture
+                      );
+                      data = fixture?.payload ?? '';
+                      break;
+                    }
+                    case 'custom': {
+                      data = customSvgData;
+                      break;
+                    }
+                    default:
+                      break;
+                  }
+                  return (
+                    <InlineSvg
+                      key={`${color}-${size}`}
+                      data={data}
+                      size={size}
+                      color={color}
+                    />
+                  );
+                })}
+              </Grid>
+            </div>
+          </Spacings.Stack>
+        ))}
+      </Spacings.Stack>
+    </Spacings.Stack>
+  );
+};
```

Comment:
This is the playground page to play around with any custom data.

![2021-07-09 12 08 53](https://user-images.githubusercontent.com/1110551/125062340-91347e80-e0ae-11eb-8bbc-4b428c393a7c.gif)